arangod/VocBase/Methods/Collections.cpp

```diff
@@ -573,15 +565,13 @@ Result Collections::drop(TRI_vocbase_t* vocbase, LogicalCollection* coll,
 Result Collections::warmup(TRI_vocbase_t& vocbase,
                            LogicalCollection const& coll) {
   ExecContext const* exec = ExecContext::CURRENT;  // disallow expensive ops
-
-  if (!exec->isSuperuser() && ServerState::readOnly()) {
+  if (!exec->canUseCollection(coll.name(), auth::Level::RW)) {
```

Comment:
Why is the warmup operation disallowed in read-only mode?
It only populates caches AFAIK. Should we really forbid it?